src/module/address/infrastructure/address.repository.ts

```diff
@@ -20,12 +19,12 @@ export class AddressRepository implements IAddressRepository {
     })
   }
 
-  async create(data: ICreateAddress): Promise<Address> {
+  async create(data: Address): Promise<Address> {
     const newAddressesArray = this.repository.create(data);
     return await this.repository.save(newAddressesArray);
   }
 
-  async update(changes: IUpdateAddress): Promise<Partial<Address>> {
+  async update(changes: Address): Promise<Partial<Address>> {
```

Comment:
This function should receive a partial Address and return an Address.
changes param should be type` Partial<Address>` and the return should be` Promise<Address>`